### PR TITLE
Tpc conditions time dependent kEff

### PR DESCRIPTION
**Pull request by mcyoren**
[comment]: <> (Please tell us something about this pull request)

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

This PR updates the handling of TPC conditions and PHGarfield configuration used by the TPC PolyClusterizer.

Main changes:

- Configure the IFC voltage correction based on the run number, so the appropriate IFC voltage settings are selected automatically for different running periods.
- Add support for time-dependent TPC GEM current conditions through `TpcConditions` (segment-by-segment).
- Move run-dependent configuration and calibration handling into PHGarfield by default.
- Load the TPC space-charge kEff values from CDB by default, while preserving explicit user overrides.
- Apply the GEM-current correction to kEff only when valid `TpcConditions` are available.
- If `TPC_CONDITIONS` is unavailable for a run, or the `TpcConditions` node is missing, reconstruction continues using the unscaled CDB kEff values and prints a warning instead of aborting.
- Handle missing or empty TPC conditions payloads.
- Add `ConditionsAvailable` to distinguish unavailable conditions from valid condition values.
- Keep manual PHGarfield/PolyClusterizer configuration available through the existing configuration interface.

The updated `TpcConditions`, `PHGarfield`, and `Tpc_PolyClusterizer` packages were tested successfully.

The modified source files were also checked with `clang-tidy` and `clang-format`.
## TODOs (if applicable)

Update CDB callibrations for kEffs. 
Add TpcConditions to Macro

## Motivation

Add time-dependent TPC condition handling to the TPC PolyClusterizer. Use CDB inputs by default while preserving explicit user configuration.

## Key changes

- Select IFC voltage distortion by run number.
- Load electric-field maps, geometry, kEff values, and field-cage settings through `PHGarfield`.
- Apply segment-level GEM-current corrections when valid `TpcConditions` are available.
- Continue reconstruction with unscaled kEff values when conditions are missing or invalid.
- Add `ConditionsAvailable` and GEM-current accessors to `TpcConditions`.
- Interpolate GEM currents and populate conditions from CDB payloads.
- Preserve manual PHGarfield and PolyClusterizer overrides.
- Add required library dependencies and the IFC voltage change run boundary.

## Potential risk areas

- Reconstruction results can change when CDB conditions become available.
- Missing conditions now produce warnings and continued processing instead of event failure.
- CDB payload interpretation and compatibility require validation.
- No thread-safety or performance assessment is provided.
- Current review severity counts are unavailable from the supplied information.

## Future improvements

- Add focused tests for empty, missing, and time-dependent condition payloads.
- Validate corrected kEff values against reference runs.
- Measure the performance impact of interpolation and CDB loading.
- Document the override precedence and run-dependent configuration.

AI-generated summaries can contain errors. Contributors should verify the implementation and physics behavior against the source code and reference data.

### CODE REVIEW

**Review by osbornjd — APPROVED** [review]
This passes the QA and similarly to your slides from today, the jenkins QA shows no change for 79513. I think if you are ready for this to go in we should merge it so that it is in the next ana build

File: offline/packages/PHGarfield/PHGarfield.cc [hunk 1]
```diff
@@ -76,7 +95,311 @@ PHGarfield::~PHGarfield()
...
+
+  m_useIFCVoltageDistortion = false;
+  m_useOFCVoltageDistortion = false;
+
+  m_ifcVoltageOffset_side0 = 0.0;
+  m_ifcVoltageOffset_side1 = 0.0;
+  m_ofcVoltageOffset_side0 = 0.0;
+  m_ofcVoltageOffset_side1 = 0.0;
+
+  if (runnumber > RunnumberRange::RUN3AUAU_IFC_V_CHANGE)
```
[the remark this answers is not in the document]
Comment (mcyoren) [reply offline/packages/PHGarfield/PHGarfield.cc:377]: 75800 is not physics run, so ">" should fine
